virttest/qemu_vm.py

```diff
@@ -2373,31 +2373,30 @@ def sort_key(dev):
                 if ats:
                     add_virtio_option("ats", ats, devices, device, dev_type)
 
-        if params.get('machine_type') == 'q35':
-            # Add extra root_port at the end of the command line only if there is
-            # free slot on pci.0, discarding them otherwise
-            func_0_addr = None
-            pcic_params = {'type': 'pcie-root-port'}
-            extra_port_num = int(params.get('pcie_extra_root_port', 0))
-            for num in range(extra_port_num):
-                try:
-                    # enable multifunction for root port
-                    port_name = "pcie_extra_root_port_%d" % num
-                    root_port = devices.pcic_by_params(port_name, pcic_params)
-                    func_num = num % 8
-                    if func_num == 0:
-                        root_port.set_param('multifunction', 'on')
-                        devices.insert(root_port)
-                        func_0_addr = root_port.get_param('addr')
-                    else:
-                        port_addr = '%s.%s' % (func_0_addr, hex(func_num))
-                        root_port.set_param('addr', port_addr)
-                        devices.insert(root_port)
-                except DeviceError:
-                    logging.warning("No sufficient free slot for extra"
-                                    " root port, discarding %d of them"
-                                    % (extra_port_num - num))
-                    break
+        # Add extra root_port at the end of the command line only if there is
```

Comment:
@PaulYuuu , would you like to help review this part.  I'm not sure if ppc platform need the extra  pice root port.

Reply:
AFAIK, pcc does not currently use pcie devices.

Reply:
@ldoktor  according @PaulYuuu 's comments, we need to ignore ppc platform.

Reply:
The "pcie_extra_root_port" is only defined for aarch64-pci and q35 so there is no need to explicitly filter this. Once/if ppc64/riscv starts using pcie and we ask for them, it should receive them without any modification to this code....

Reply:
@ldoktor  some tests configured  'pcie-extra_root_port'  but these case are not x86/x86_64 only tests.  so I suggest to keep original code block:

xutian$ grep 'pcie_extra_root_port'  -R |grep cfg
qemu/tests/cfg/smbios.cfg:    pcie_extra_root_port = 0
qemu/tests/cfg/block_hotplug_in_pause.cfg:                pcie_extra_root_port = 2
qemu/tests/cfg/virtio_console.cfg:                                pcie_extra_root_port = 4
qemu/tests/cfg/rng_hotplug.cfg:                pcie_extra_root_port = 4
qemu/tests/cfg/block_hotplug.cfg:                pcie_extra_root_port = 2
qemu/tests/cfg/seabios_hotplug_unplug.cfg:        pcie_extra_root_port = 2
qemu/tests/cfg/nic_hotplug.cfg:                pcie_extra_root_port = 2
qemu/tests/cfg/nic_hotplug.cfg:                    pcie_extra_root_port = 0
qemu/tests/cfg/multi_nics_hotplug.cfg:        pcie_extra_root_port = 4

Reply:
The original code does not work as there are other archs that require the pcie_extra_ports (for now just arm and q35, but this should expand with riscv in the future and perhaps pseries will start utilize pcie aswell...).

How about looking at busses and if the machine contains PCIE bus add the extra root ports?